src/Core/Form/ChoiceProvider/OrderStateByIdChoiceProvider.php

```diff
@@ -51,31 +53,46 @@ final class OrderStateByIdChoiceProvider implements FormChoiceProviderInterface,
      */
     private $colorBrightnessCalculator;
 
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
     /**
      * @param int $languageId language ID
      * @param OrderStateDataProviderInterface $orderStateDataProvider
+     * @param ColorBrightnessCalculator $colorBrightnessCalculator
+     * @param TranslatorInterface $translator
      */
     public function __construct(
         $languageId,
         OrderStateDataProviderInterface $orderStateDataProvider,
-        ColorBrightnessCalculator $colorBrightnessCalculator
+        ColorBrightnessCalculator $colorBrightnessCalculator,
+        TranslatorInterface $translator
     ) {
         $this->languageId = $languageId;
         $this->orderStateDataProvider = $orderStateDataProvider;
         $this->colorBrightnessCalculator = $colorBrightnessCalculator;
+        $this->translator = $translator;
     }
 
     /**
      * Get order state choices.
      *
+     * @param array $options
+     *
      * @return array
      */
-    public function getChoices()
+    public function getChoices(array $options = [])
     {
         $orderStates = $this->orderStateDataProvider->getOrderStates($this->languageId);
         $choices = [];
 
         foreach ($orderStates as $orderState) {
+            if ($orderState['deleted'] == 1 && (empty($options['current_state']) || $options['current_state'] != $orderState['id_order_state'])) {
+                continue;
+            }
+            $orderState['name'] .= $orderState['deleted'] == 1 ? ' ' . $this->translator->trans('(deleted)', [], '') : '';
             $choices[$orderState['name']] = $orderState['id_order_state'];
```

Comment:
@LouiseBonnard here we will display the mention `(deleted)` with parenthesis, following the order status name, if the status has been deleted. Do you have any idea about what domain we should use ?

Reply:
@Junebyun is to one to help you get the proper domain. ;-)

Reply:
@matthieu-rolland Hi! If I got it right the _(deleted)_ message would appear in the BO, correct?

Reply:
yes 👍 we have a selector with a list of order statuses (ex: "waiting for payment"), and we will add `(deleted)` when the currently selected status is deleted so that the user knows that once he selects another order status this one will disappear.

Reply:
@matthieu-rolland Hi! You can put it in the _Admin.Global_ domain :)

Reply:
@Junebyun thanx 👍 done